### PR TITLE
Add option to define mandatory configuration properties for plugins

### LINKED ISSUE #512: Add option for plugins to declare required properties
Currently for Openstack, certain configuration properties must be available for it work by default. 
Ensure that each plugin defines the list of properties that it needs for it to work. 

This is needed to avoid everything to be configured as credential.

### PR DESCRIPTION
**Pull request by swaroopar**
fixes #512

### CODE REVIEW

File: modules/orchestrator/src/main/java/org/eclipse/xpanse/modules/orchestrator/PluginManager.java [hunk 1]
```diff
@@ -23,21 +27,27 @@
 @Component
 public class PluginManager {
 
-    private final Map<Csp, OrchestratorPlugin> pluginMap = new ConcurrentHashMap<>();
+    @Getter
```
[the remark this answers is not in the document]
Comment (swaroopar) [reply modules/orchestrator/src/main/java/org/eclipse/xpanse/modules/orchestrator/PluginManager.java:30]: @sonatype-lift ignore

File: plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/monitor/keystone/KeystoneManager.java [hunk 2]
```diff
@@ -75,7 +84,7 @@ public void authenticate(AbstractCredentialInfo credential) {
                         credentialVariable.getName())) {
                     proxyPort = credentialVariable.getValue();
                 }
-                if (OpenstackEnvironmentConstants.SERVICE_TENANT.equals(
+                if (OpenstackEnvironmentConstants.SERVICE_PROJECT.equals(
```
[the remark this answers is not in the document]
Comment (swaroopar) [reply plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/monitor/keystone/KeystoneManager.java:88]: @sonatype-lift ignore

File: plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/monitor/keystone/KeystoneManager.java [hunk 3]
```diff
@@ -55,7 +64,7 @@ public void authenticate(AbstractCredentialInfo credential) {
                         credentialVariable.getName())) {
                     password = credentialVariable.getValue();
                 }
-                if (OpenstackEnvironmentConstants.TENANT.equals(
+                if (OpenstackEnvironmentConstants.PROJECT.equals(
```
[the remark this answers is not in the document]
Comment (swaroopar) [reply plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/monitor/keystone/KeystoneManager.java:68]: @sonatype-lift ignore

File: plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/monitor/keystone/KeystoneManager.java [hunk 4]
```diff
@@ -75,7 +85,7 @@ public void authenticate(AbstractCredentialInfo credential) {
```
[the remark this answers is not in the document]
Comment (swaroopar) [reply plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/monitor/keystone/KeystoneManager.java]: @sonatype-lift ignore

File: plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/monitor/keystone/KeystoneManager.java [hunk 5]
```diff
@@ -55,7 +65,7 @@ public void authenticate(AbstractCredentialInfo credential) {
```
[the remark this answers is not in the document]
Comment (swaroopar) [reply plugins/openstack/src/main/java/org/eclipse/xpanse/plugins/openstack/monitor/keystone/KeystoneManager.java]: @sonatype-lift ignore

**Review by baixinsui — APPROVED** [review]
LGTM

**Review by iskey — APPROVED** [review]
LGTM

**Pull request merged**